lib/deprecated/structs/ApplyOptions.js

```diff
@@ -0,0 +1,3 @@
+module.exports = function() {
+  this.flags = 0;
```

Comment:
Only field 'flags' needed here? why not the rest of the fields of the structure in https://libgit2.org/libgit2/#HEAD/type/git_apply_options

Reply:
Those fields don't really benefit from having a default value. It's important that fields that are numbers are built out with a default value so that users who are depending on those items being numbers continue to work correctly.

The best example of this is somewhere in our codebase we do: `diffOptions.flags |= NodeGit.Diff.IGNORE_WHITESPACE`. But this would not work if flags is not already a number, as `undefined | 5` produces `NaN`.

If we compare that to any of the other fields that are not numbers, the default values for these are `null`. In JS we're basically just expressing `null` as `undefined` by the field not being on the object.